app/controllers/ErrorPagesHandler.scala

```diff
@@ -29,19 +29,23 @@ import uk.gov.hmrc.tai.model.domain.Employment
 import uk.gov.hmrc.tai.util.constants.TaiConstants
 import uk.gov.hmrc.tai.util.constants.TaiConstants._
 import uk.gov.hmrc.urls.Link
+import views.html.{error_no_primary, error_template_noauth}
 
 import scala.concurrent.Future
 
 trait ErrorPagesHandler {
 
+  val error_template_noauth: error_template_noauth
+  val error_no_primary: error_no_primary
```

Comment:
If you make ErrorPagesHandler a class and inject it in to the controllers that use it, I think it'll tidy a lot of this code up as we've ended up (due to `TaiBaseController` extending `ErrorPagesHandler`)  everything being an `ErrorPagesHandler` when it really does not need to be.